Native Picocli build breaks SSE client

### Describe the bug

When using Picocli and JAX-RS to consume SSEs, everything works well, until the app is built into a native executable.

The following exception is thrown, indicating that JAX-RS classes can't be found:

```
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2023-10-11 10:25:39,671 INFO  [io.quarkus] (main) quarkus-playground-client 1.0 native (powered by Quarkus 3.2.0.Final) started in 0.009s. Listening on:
2023-10-11 10:25:39,672 INFO  [io.quarkus] (main) Profile prod activated.
2023-10-11 10:25:39,672 INFO  [io.quarkus] (main) Installed features: [cdi, picocli, rest-client, rest-client-jsonb, resteasy, resteasy-jsonb, smallrye-context-propagation, vertx]
running test run()
java.lang.RuntimeException: java.lang.ClassNotFoundException: Provider for jakarta.ws.rs.sse.SseEventSource.Builder cannot be found
	at jakarta.ws.rs.sse.SseEventSource$Builder.newBuilder(SseEventSource.java:140)
	at jakarta.ws.rs.sse.SseEventSource.target(SseEventSource.java:232)
	at com.sebastian_daschner.client.SseClient.someMethod(SseClient.java:35)
	at com.sebastian_daschner.client.SseClient_ClientProxy.someMethod(Unknown Source)
	at com.sebastian_daschner.client.TestCommand.run(TestCommand.java:26)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2026)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at io.quarkus.picocli.runtime.PicocliRunner$EventExecutionStrategy.execute(PicocliRunner.java:26)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at io.quarkus.picocli.runtime.PicocliRunner.run(PicocliRunner.java:40)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:132)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:71)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:44)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
Caused by: java.lang.ClassNotFoundException: Provider for jakarta.ws.rs.sse.SseEventSource.Builder cannot be found
	at jakarta.ws.rs.sse.FactoryFinder.find(FactoryFinder.java:163)
	at jakarta.ws.rs.sse.SseEventSource$Builder.newBuilder(SseEventSource.java:125)
	... 18 more
2023-10-11 10:25:39,678 INFO  [io.quarkus] (main) quarkus-playground-client stopped in 0.002s
```

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

Reproducer here: https://github.com/sdaschner/quarkus-playground/tree/36402-sse-native-cli

```
cd server/
mvn package
java -jar target/quarkus-app/quarkus-run.jar

# different shell
cd client/
mvn package
java -jar target/quarkus-app/quarkus-run.jar --client
# works

mvn package -Dquarkus.package.type=native
target/client-runner --client
# fails with exception
```


### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

OpenJDK Runtime Environment Temurin-18.0.1+10 (build 18.0.1+10)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

3.4.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java
+++ b/extensions/resteasy-classic/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java
@@ -19,6 +19,7 @@ import jakarta.ws.rs.Path;
 import jakarta.ws.rs.client.ClientRequestFilter;
 import jakarta.ws.rs.client.ClientResponseFilter;
 import jakarta.ws.rs.ext.Providers;
+import jakarta.ws.rs.sse.SseEventSource;
 
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.config.ConfigProvider;
@@ -48,6 +49,7 @@ import org.jboss.resteasy.microprofile.client.DefaultResponseExceptionMapper;
 import org.jboss.resteasy.microprofile.client.RestClientProxy;
 import org.jboss.resteasy.microprofile.client.async.AsyncInterceptorRxInvokerProvider;
 import org.jboss.resteasy.microprofile.client.publisher.MpPublisherMessageBodyReader;
+import org.jboss.resteasy.plugins.providers.sse.client.SseEventSourceImpl;
 import org.jboss.resteasy.spi.ResteasyConfiguration;
 
 import io.quarkus.arc.BeanDestroyer;
@@ -122,8 +124,11 @@ class RestClientProcessor {
     }
 
     @BuildStep
-    void setupClientBuilder(BuildProducer<NativeImageResourceBuildItem> resources) {
+    void setupClientBuilder(BuildProducer<NativeImageResourceBuildItem> resources,
+            BuildProducer<ServiceProviderBuildItem> serviceProviders) {
         resources.produce(new NativeImageResourceBuildItem("META-INF/services/jakarta.ws.rs.client.ClientBuilder"));
+        serviceProviders.produce(new ServiceProviderBuildItem(SseEventSource.Builder.class.getName(),
+                SseEventSourceImpl.SourceBuilder.class.getName()));
     }
 
     @BuildStep

--- a/extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java
+++ b/extensions/resteasy-reactive/jaxrs-client-reactive/deployment/src/main/java/io/quarkus/jaxrs/client/reactive/deployment/JaxrsClientReactiveProcessor.java
@@ -65,6 +65,7 @@ import jakarta.ws.rs.core.MultivaluedMap;
 import jakarta.ws.rs.core.UriBuilder;
 import jakarta.ws.rs.ext.ParamConverter;
 import jakarta.ws.rs.ext.ParamConverterProvider;
+import jakarta.ws.rs.sse.SseEventSource;
 
 import org.eclipse.microprofile.config.Config;
 import org.eclipse.microprofile.config.ConfigProvider;
@@ -86,6 +87,7 @@ import org.jboss.resteasy.reactive.client.impl.AsyncInvokerImpl;
 import org.jboss.resteasy.reactive.client.impl.ClientBuilderImpl;
 import org.jboss.resteasy.reactive.client.impl.ClientImpl;
 import org.jboss.resteasy.reactive.client.impl.MultiInvoker;
+import org.jboss.resteasy.reactive.client.impl.SseEventSourceBuilderImpl;
 import org.jboss.resteasy.reactive.client.impl.StorkClientRequestFilter;
 import org.jboss.resteasy.reactive.client.impl.UniInvoker;
 import org.jboss.resteasy.reactive.client.impl.WebTargetImpl;
@@ -226,6 +228,8 @@ public class JaxrsClientReactiveProcessor {
         serviceProviders.produce(new ServiceProviderBuildItem(ClientBuilder.class.getName(),
                 ClientBuilderImpl.class.getName()));
 
+        serviceProviders.produce(new ServiceProviderBuildItem(SseEventSource.Builder.class.getName(),
+                SseEventSourceBuilderImpl.class.getName()));
     }
 
     @BuildStep

